File: 0804-unique-morse-code-words/0804-unique-morse-code-words.py
import string
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        morse= [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        alphabet = string.ascii_lowercase
        
        alpha_to_morse = {}
        for i in range(len(alphabet)):
            alpha_to_morse[alphabet[i]] = morse[i]
        
        res = set()
        
        for word in words:
            temp = ""
            for char in word:
                temp += alpha_to_morse[char]
            res.add(temp)
        
        return len(res)

File: 0804-unique-morse-code-words/0804-unique-morse-code-words.ts
function uniqueMorseRepresentations(words: string[]): number {
    const alphaToMorse = {
        "a": ".-",
        "b": "-...",
        "c": "-.-.",
        "d": "-..",
        "e": ".",
        "f": "..-.",
        "g": "--.",
        "h": "....",
        "i": "..",
        "j": ".---",
        "k": "-.-",
        "l": ".-..",
        "m": "--",
        "n": "-.",
        "o": "---",
        "p": ".--.",
        "q": "--.-",
        "r": ".-.",
        "s": "...",
        "t": "-",
        "u": "..-",
        "v": "...-",
        "w": ".--",
        "x": "-..-",
        "y": "-.--",
        "z": "--.."
    }
    
    let res = 0;
    const seen = new Set<string>();
    
    for (const word of words) {
        let newWord = ''
        for (const char of word) {
            newWord += alphaToMorse[char];
        }
        if (!seen.has(newWord)) {
            seen.add(newWord);
            res++;
        }
    }
    
    return res;
};